pynumaflow/sourcer/_dtypes.py

```diff
@@ -0,0 +1,166 @@
+
+    def __init__(self, offset: bytes, partition_id: str):
+        self._offset = offset
+        self._partition_id = partition_id
+
+    @property
+    def as_dict(self):
+        return {"offset": self._offset, "partition_id": self._partition_id}
+
+    @property
+    def offset(self) -> bytes:
+        return self._offset
+
+    @property
+    def partition_id(self) -> str:
+        return self._partition_id
+
+
+@dataclass(init=False)
+class Message:
+    """
+    Basic datatype for data passing to the next vertex/vertices.
+
+    Args:
+        payload: data in bytes
+        offset: the offset of the datum.
+        event_time: event time of the message, usually extracted from the payload.
+        keys: []string keys for vertex (optional)
+    """
+
+    __slots__ = ("_payload", "_offset", "_event_time", "_keys")
+
+    _payload: bytes
+    _offset: Offset
+    _event_time: datetime
+    _keys: list[str]
+
+    def __init__(
+        self, payload: bytes, offset: Offset, event_time: datetime, keys: list[str] = None
+    ):
+        """
+        Creates a Message object to send value to a vertex.
+        """
+        self._payload = payload
+        self._offset = offset
+        self._event_time = event_time
+        self._keys = keys or []
+
+    @property
+    def payload(self) -> bytes:
+        return self._payload
+
+    @property
+    def keys(self) -> list[str]:
+        return self._keys
+
+    @property
+    def offset(self) -> Offset:
+        return self._offset
+
+    @property
+    def event_time(self) -> datetime:
+        return self._event_time
+
+
+@dataclass(init=False)
+class Datum:
+    """
+    Class to define the important information for the event.
+    Args:
+        num_records: the number of records to read.
+        timeout_in_ms: the request timeout in milliseconds.
+    >>> # Example usage
+    >>> from pynumaflow.sourcer import Datum
+    >>> from datetime import datetime, timezone
+    >>> datum = Datum(num_records=10, timeout_in_ms=1000)
+    """
+
+    __slots__ = ("_num_records", "_timeout_in_ms")
+
+    _num_records: int
+    _timeout_in_ms: int
+
+    def __init__(
+        self,
+        num_records: int,
+        timeout_in_ms: int,
+    ):
+        if not isinstance(num_records, int):
+            raise TypeError(f"Wrong data type: {type(num_records)} for Datum.num_records")
+        self._num_records = num_records
+        if not isinstance(timeout_in_ms, int):
+            raise TypeError(f"Wrong data type: {type(timeout_in_ms)} for Datum.timeout_in_ms")
+        self._timeout_in_ms = timeout_in_ms
+
+    @property
+    def num_records(self) -> int:
+        """Returns the num_records of the event"""
+        return self._num_records
+
+    @property
+    def timeout_in_ms(self) -> int:
+        """Returns the timeout_in_ms of the event."""
+        return self._timeout_in_ms
+
+
+@dataclass(init=False)
+class AckRequest:
+    """
+    Class to define the ack information for the event.
```

Comment:
Please follow `ReadRequest` above to write the class description. Same for the other classes.